xmppserver/src/main/java/org/jivesoftware/openfire/Connection.java

```diff
@@ -16,24 +16,38 @@
 
 package org.jivesoftware.openfire;
 
+import org.dom4j.Namespace;
 import org.jivesoftware.openfire.auth.UnauthorizedException;
 import org.jivesoftware.openfire.session.LocalSession;
 import org.jivesoftware.openfire.spi.ConnectionConfiguration;
 import org.xmpp.packet.Packet;
 import org.xmpp.packet.StreamError;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.io.Closeable;
 import java.net.UnknownHostException;
 import java.security.cert.Certificate;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * Represents a connection on the server.
  *
  * @author Iain Shigeoka
  */
 public interface Connection extends Closeable {
-    
+
+    /**
+     * When a connection is used to transmit an XML data, the root element of that data can define XML namespaces other
+     * than the ones that are default (eg: 'jabber:client', 'jabber:server', etc). For an XML parser to be able to parse
+     * stanzas or other elements that are defined in that namespace (eg: are prefixed), these namespaces are recorded
+     * here.
+     *
+     * @see <a href="https://igniterealtime.atlassian.net/browse/OF-2556">Issue OF-2556</a>
+     */
+    Set<Namespace> additionalNamespaces = new HashSet<>();
```

Comment:
Given your current implementation, this could be a final field. If it's not supposed to be a final, why not simply re-initialise it in the setter method?

Reply:
IntelliJ tells me that the modifier 'final' is redundant on interface fields.

I'm not sure if this should be an interface field though. Maybe this needs to go in an abstract implementation? I am not used to using interfaces this way...